app/http.php

```diff
@@ -0,0 +1,1039 @@
+    }
+
+    $callables = [];
+
+    foreach ($functionsToRemove as $container) {
+        $callables[] = function () use ($container, $orchestrationPool) {
+            try {
+                $connection = $orchestrationPool->pop();
+                $orchestration = $connection->getResource();
+                $orchestration->remove($container->getId(), true);
+                Console::success('Removed container ' . $container->getName());
+            } catch (\Throwable $th) {
+                Console::error('Failed to remove container: ' . $container->getName());
+                Console::error($th);
+            } finally {
+                isset($connection) && $orchestrationPool->push($connection);
+            }
+        };
+    }
+
+    /** @phpstan-ignore-next-line */
+    Swoole\Coroutine\batch(
+        $callables
+    );
+
+    Console::success('Done.');
+}
+
+App::post('/v1/runtimes')
+    ->desc("Create a new runtime server")
+    ->param('runtimeId', '', new Text(64), 'Unique runtime ID.')
+    ->param('image', '', new Text(128), 'Base image name of the runtime.')
+    ->param('source', '', new Text(0), 'Path to source files.')
+    ->param('entrypoint', '', new Text(256), 'Entrypoint of the code file.')
+    ->param('destination', '', new Text(0), 'Destination folder to store build files into.', true)
+    ->param('variables', [], new Assoc(), 'Environment Variables required for the build.', true)
+    ->param('commands', [], new ArrayList(new Text(1024), 100), 'Commands required to build the container. Maximum of 100 commands are allowed, each 1024 characters long.', true)
+    ->param('workdir', '', new Text(256), 'Working directory.', true)
+    ->param('remove', false, new Boolean(), 'Remove a runtime after execution.', true)
+    ->inject('orchestrationPool')
+    ->inject('activeRuntimes')
+    ->inject('response')
+    ->action(function (string $runtimeId, string $image, string $source, string $entrypoint, string $destination, array $variables, array $commands, string $workdir, bool $remove, Pool $orchestrationPool, Table $activeRuntimes, Response $response) {
+        $activeRuntimeId = $runtimeId; // Used with Swoole table (key)
+        $runtimeId = System::getHostname() . '-' . $runtimeId; // Used in Docker (name)
+
+        $runtimeHostname = \uniqid();
+
+        if ($activeRuntimes->exists($activeRuntimeId)) {
+            if ($activeRuntimes->get($activeRuntimeId)['status'] == 'pending') {
+                throw new \Exception('A runtime with the same ID is already being created. Attempt a execution soon.', 500);
+            }
+
+            throw new Exception('Runtime already exists.', 409);
+        }
+
+        $container = [];
+        $containerId = '';
+        $stdout = '';
+        $stderr = '';
+        $startTimeUnix = \microtime(true);
+        $endTimeUnix = 0;
+        $connection = $orchestrationPool->pop();
+        $orchestration = $connection->getResource();
```

Comment:
It might be much easier and cleaner to just directly get the orchestrator instance inside the dependency injection callback in line 112 with the pop() method followed by the getResource() method. You can also see an example on the dbPools branch: https://github.com/appwrite/appwrite/blob/feat-db-pools/app/init.php#L1045-L1056

Reply:
Hmm.. I see I need to create resource. But how is it then reclaimed? Pools still confuse me a little. How would I reclaim this specific connection in shutdown?

Reply:
Logic implemented. There is now a resource for a connection, and it is reclaimed after a request (in shutdown)